### ISSUE TITLE
Adapt or remove disabled tests checking loading and running bytecode files from PUC-Rio Lua 5.1 test suite in main.lua

### ISSUE THREAD

**Issue by Buristan** [tarantool/tarantool#5691]
This ticket is a logical continuation of gh-4473.

If run  PUC-Rio Lua 5.1 test suite by LuaJIT (HEAD at the moment: [`1672bdc`](https://github.com/tarantool/luajit/commit/1672bdc0ffbee9f32fadb6943909d6af7eb9a7b1)) (not Tarantool) it fails with the following error:
```
/home/burii/builds_workspace/luajit/gh-4473-adopt-luavela-test-suites/src/luajit: cannot load malformed bytecode
/home/burii/builds_workspace/luajit/gh-4473-adopt-luavela-test-suites/src/luajit: main.lua:42: assertion failed!
stack traceback:
        [C]: in function 'assert'
        main.lua:42: in function 'RUN'
        main.lua:158: in function 'dofile'
        ...adopt-luavela-test-suites/test/PUC-Lua-5.1-tests/all.lua:69: in main chunk
        [C]: at 0x55957cbdff96
```
caused by this test chunk (assuming that all previous failing test chunks non relating to this error are disabled):
```lua
prepfile("#comment with a binary file\n"..string.dump(loadstring("print(1)")))
RUN("lua %s > %s", prog, out)
checkout("1\n")
```

Loading bytecode with an extra header (BOM or "#") is disabled for security reasons since LuaJIT-2.0.0-beta10.
For more information see comment for `lj_lex_setup()` in <src/lj_lex.c>.
Also see commit 53a285c0c3544ff5dea7c67b741c3c2d06d22b47 (Disable loading bytecode with an extra header (BOM or #!).).

**Comment by Buristan** [comment]
This test chunk can be adapted to LuaJIT behavior like this:
```diff
diff --git a/test/PUC-Lua-5.1-tests/main.lua b/test/PUC-Lua-5.1-tests/main.lua
index 77313a9..b0688ee 100644
--- a/test/PUC-Lua-5.1-tests/main.lua
+++ b/test/PUC-Lua-5.1-tests/main.lua
@@ -155,16 +155,11 @@ RUN("lua %s", prog)
 -- Also see commit 53a285c0c3544ff5dea7c67b741c3c2d06d22b47
 -- (Disable loading bytecode with an extra header (BOM or #!).).
 -- See also https://github.com/tarantool/tarantool/issues/5691.
---[==[
-prepfile("#comment with a binary file\n"..string.dump(loadstring("print(1)")))
+-- Test is adapted to the behaviour of LuaJIT.
+prepfile(string.dump(loadstring("print(1)")))
 RUN("lua %s > %s", prog, out)
 checkout("1\n")
 
-prepfile("#comment with a binary file\r\n"..string.dump(loadstring("print(1)")))
-RUN("lua %s > %s", prog, out)
-checkout("1\n")
---]==]
-
 -- close Lua with an open file
 prepfile(string.format([[io.output(%q); io.write('alo')]], out))
 RUN("lua %s", prog)
```

**Comment by igormunkin** [comment]
The issue is resolved in scope of tarantool/luajit@ba560c1. The submodule was updated in tarantool in 2.10.0-beta2-121-gbc1deaff3 (bc1deaf), 2.8.3-48-ga08234892 (a082348), 1.10.12-41-gef9c0d6af (ef9c0d6).
